### PR TITLE
MARS-9: Display dates for the Kennedy Space Center on homepage

### PR DESCRIPTION
**Pull request by leedoughty**
**Changes**
- Display dates for the Kennedy Space Center on homepage
- Display the time zone offset (in hours) of the user's time zone compared to the Kennedy Space Center.

Screenshot 2020-11-13 at 09 00 04

### CODE REVIEW

File: public/script.js [hunk 1]
```diff
@@ -1,5 +1,44 @@
-const earthDateParagraph = document.querySelector("#earth-date");
 const earthDate = new Date();
-const dateText = document.createTextNode(earthDate.toLocaleString());
 
-earthDateParagraph.appendChild(dateText);
+function displayCurrentEarthDate() {
+  const earthDateParagraph = document.querySelector("#earth-date");
+  const dateText = document.createTextNode(earthDate.toLocaleString());
+
+  earthDateParagraph.appendChild(dateText);
```
Comment (Kenneth-Siu) [inline public/script.js]: One thing that could make it a little bit lighter on the mental overhead is to chain together the `querySelector` and the `appendChild`.

Chaining things together (...within reason) in general can make it a little lower on the mental overhead as it reduces the number of variables floating about.

File: public/script.js [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+  );
+  const behindKennedy = document.createTextNode(
+    " hours behind of Kennedy Space Center)"
+  );
+
+  if (timeBetweenKennedyAndUtc > 0) {
+    kennedyOffsetTime.appendChild(aheadOfKennedy);
+  } else {
+    kennedyOffsetTime.appendChild(behindKennedy);
+  }
```
Comment (Kenneth-Siu) [inline public/script.js]: Could use string interpolation to build the string, and then put it all in as a child text node in one go?

**Review by Kenneth-Siu — COMMENTED** [review]
Neatly laid out with well-thought-out variable and function names!

I've left a few ideas for how to make it a bit lighter of a read.

**Comment by leedoughty** [comment]
> Neatly laid out with well-thought-out variable and function names!
> 
> I've left a few ideas for how to make it a bit lighter of a read.

Cool :) will take a look at those suggestions now

File: public/script.js (hunk 2 above)
Comment (Kenneth-Siu) [reply public/script.js]: I think you could combine both ahead and behind to be just the one string!

File: public/script.js [hunk 3]
```diff
@@ -1,5 +1,38 @@
-const earthDateParagraph = document.querySelector("#earth-date");
 const earthDate = new Date();
-const dateText = document.createTextNode(earthDate.toLocaleString());
 
-earthDateParagraph.appendChild(dateText);
+function displayCurrentEarthDate() {
+  const dateText = document.createTextNode(earthDate.toLocaleString());
+  document.querySelector("#earth-date").appendChild(dateText);
```
Comment (Kenneth-Siu) [inline public/script.js:5]: I often find it's a bit more readable if the extra chains are on separate lines (and aligned nicely)

```
    document.querySelector("#earth-date")
            .appendChild(dateText);
```

(And same for the rest of the chaining in this file)

**Review by Kenneth-Siu — CHANGES_REQUESTED** [review]

File: public/script.js (hunk 2 above)
Comment (Kenneth-Siu) [reply public/script.js]: Something like `${time} hours ${isAhead ? "ahead" : "behind"} of KSC`

(...but done properly)

Comment (leedoughty) [reply public/script.js]: Sounds good, will take a look at that now

File: public/script.js (hunk 3 above)
Comment (leedoughty) [reply public/script.js:5]: My Prettier VS Code extension (which formats on save) doesn't allow me to change the first two instances of the chaining. I am planning to setup Prettier and ESLint to run as a commit hook in this repo in the future though (and I will add that rule to move chains onto separate lines when I set that up) do you think I should do this as part of this PR, or as a separate PR? 🤔

**Review by Kenneth-Siu — APPROVED** [review]
Excellent, looks good! 👍

**Pull request merged**